ocs_ci/utility/kms.py

```diff
@@ -1797,7 +1802,284 @@ def cleanup(self):
+            )
+
+        if self.azure_kms_connection_name not in csi_kms_configmap.data["data"]:
+            raise ValueError(
+                f"Azure Key vault connection {self.azure_kms_connection_name} not exists."
+            )
+
+        return True
+
+    def create_azure_kv_secrets(self, prefix="azure-ocs-"):
+        """
+        Creates Azure KV secrets.
+        """
+        secret_name = gen_alphanumeric(length=18, start=prefix).lower()
+        client_secret = templating.load_yaml(constants.AZURE_CLIENT_SECRETS)
+
+        client_secret["metadata"]["name"] = secret_name
+        client_secret["metadata"]["namespace"] = self.namespace
+        client_secret["data"]["CLIENT_CERT"] = base64.b64encode(
+            self.azure_kv_certificate.encode()
+        ).decode()
+        logger.info(f"Creating a Azure Secret : {secret_name}")
+        self.create_resource(client_secret, prefix=prefix)
+        return secret_name
+
+    def create_azure_kv_csi_kms_connection_details(self):
+        """
+        Create Azure specific csi-kms-connection-details
+        configmap resource
+        """
+
+        # Check is already configmap exists
+        csi_kms_configmap = ocp.OCP(
+            kind="ConfigMap",
+            resource_name=constants.AZURE_KV_CSI_CONNECTION_DETAILS,
+            namespace=self.namespace,
+        )
+
+        # Create a Connection data.
+        azure_conn = self.conn_data
+        azure_conn["AZURE_CERT_SECRET_NAME"] = self.create_azure_kv_secrets(
+            prefix="azure-csi-"
+        )
+
+        if not csi_kms_configmap.is_exist():
+            logger.info(
+                f"Creating Configmap {constants.AZURE_KV_CSI_CONNECTION_DETAILS}"
+            )
+
+            csi_kms_conn_details = templating.load_yaml(
+                constants.AZURE_CSI_KMS_CONNECTION_DETAILS
+            )
+
+            # Updating Templet data.
+            csi_kms_conn_details["data"] = {
+                self.azure_kms_connection_name: json.dumps(azure_conn)
+            }
+
+            csi_kms_conn_details["metadata"]["namespace"] = self.namespace
+            self.create_resource(csi_kms_conn_details, prefix="csiazureconn")
+        else:
+            logger.info(
+                f"Adding Azure connection to existing ConfigMap {constants.AZURE_KV_CSI_CONNECTION_DETAILS}"
+            )
+            param = (
+                f'[{{"op": "add", "path": "/data/{self.azure_kms_connection_name}", '
+                f'"value": "{json.dumps(azure_conn)}"}}]'
+            )
+            csi_kms_configmap.patch(params=param, format_type="merge")
+
+        # verifying ConfigMap is created or not.
+        try:
+            self.is_azure_kv_connection_exists()
+            return True
+        except ValueError as ex:
+            logger.info(
+                f"Error Creating resource {constants.AZURE_KV_CSI_CONNECTION_DETAILS} : {ex}"
+            )
+            return False
```

Comment:
as commented above, self.is_azure_kv_connection_exists() should return True or False and not throw an exception. With such a change, you will not need to try and except here